### PR TITLE
CI for UI Tests Infrastructure Update

### PR DESCRIPTION
**Pull request by PiotrKarczmarz**
CLOSE https://linear.app/sourcegraph/issue/CODY-2816/ci-for-visual-studio-integration-tests-based-on-vsixtesting

UI Tests infrastructure (`Cody.VisualStudio.Tests`) updated to allow running them, not only locally which were running fine from the very beginning, but also using GitHub Actions.

### Fixed threading issues only seen in GitHub Actions:
 - VsixTesting runner exception: `Call was Rejected By Callee` ([more about it](https://learn.microsoft.com/en-us/previous-versions/ms228772(v=vs.140)?redirectedfrom=MSDN)), caused PlaywrightTestsBase.InitializeAsync() was called multiple times, which internally tries to call VS API,  when XUnit was creating instances of all involved test classes
 - Removed calls to VS API from TestBase constructor, which have the same effect as above

### CODE REVIEW

File: src/Cody.VisualStudio.Tests/ChatLoggedBasicTests.cs [hunk 1]
```diff
@@ -1,57 +1,83 @@
...
 namespace Cody.VisualStudio.Tests
 {
     public class ChatLoggedBasicTests: PlaywrightInitializationTests
     {
-        [VsFact(Version = VsVersion.VS2022)]
+        public ChatLoggedBasicTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        //[VsFact(Version = VsVersion.VS2022)]
```
Comment (abeatrix) [inline src/Cody.VisualStudio.Tests/ChatLoggedBasicTests.cs:16]: what is this for?

File: src/build.cake [hunk 2]
```diff
@@ -153,11 +153,11 @@ Task("Tests")
 		Verbosity = Verbosity.Minimal
 	});
 
-	DotNetTest("./Cody.VisualStudio.Tests/bin/Debug/Cody.VisualStudio.Tests.dll", new DotNetTestSettings 
-    {
-        NoBuild = true,
-        NoRestore = true
-    });
+//	DotNetTest("./Cody.VisualStudio.Tests/bin/Debug/Cody.VisualStudio.Tests.dll", new DotNetTestSettings
```
Comment (abeatrix) [inline src/build.cake:156]: Do we need to keep this comment?

**Review by abeatrix — APPROVED** [review]
Looks great in the CI! Let's document how dev can add a new test and what steps devs should take when the CI fail in our [CONTRIBUTING](https://github.com/sourcegraph/cody-vs/blob/main/CONTRIBUTING.md) docs in a follow up. Will create a linear issue for this, thank you so much!

File: src/build.cake (hunk 2 above)
Comment (PiotrKarczmarz) [reply src/build.cake:156]: Yes, I will change it later, when I will finish the next part of CI updates.

File: src/Cody.VisualStudio.Tests/ChatLoggedBasicTests.cs (hunk 1 above)
Comment (PiotrKarczmarz) [reply src/Cody.VisualStudio.Tests/ChatLoggedBasicTests.cs:16]: This is for XUnit logger, which is passing calls to our Logger. Thanks to it we have nice, detailed logs https://github.com/sourcegraph/cody-vs/actions/runs/10814320609/job/30000502786

image

**Pull request merged**